__tests__/basic-auth-middleware.test.js

```diff
@@ -0,0 +1,63 @@
+'use strict';
+
+require('@code-fellows/supergoose');
+const middleware = require('../src/middleware/basic');
+const Users = require('../src/models/users/users');
+
+let users = {
+  admin: { username: 'admin', password: 'password', role: 'admin' },
+};
+
+// Pre-load our database with fake users
+beforeAll(async (done) => {
+  await new Users(users.admin).save();
+  done();
+});
+
+describe('Auth Middleware', () => {
+
+  // admin:password: YWRtaW46cGFzc3dvcmQ=
+  // admin:foo: YWRtaW46Zm9v
+
+  // Mock the express req/res/next that we need for each middleware call
+  const req = {};
+  const res = {
+    status: jest.fn(() => res),
+    send: jest.fn(() => res)
+  }
+  const next = jest.fn();
+
+  describe('user authentication', () => {
+
+    it('fails a login for a user (admin) with the incorrect basic credentials', () => {
+
+      // Change the request to match this test case
+      req.headers = {
+        authorization: 'Basic YWRtaW46Zm9v',
+      };
+
+      return middleware(req, res, next)
+        .then(() => {
+          expect(next).not.toHaveBeenCalled();
+          expect(res.status).toHaveBeenCalledWith(403);
+        });
+
+    }); // it()
+
+    it('logs in an admin user with the right credentials', () => {
+
+      // Change the request to match this test case
+      req.headers = {
+        authorization: 'Basic YWRtaW46cGFzc3dvcmQ=',
+      };
+
+      return middleware(req, res, next)
+        .then(() => {
+          expect(next).toHaveBeenCalledWith();
+        });
+
+    }); // it()
+
+  });
+
```

Comment:
These tests look really great, I would just like to see tests for v1 or v2 routes